shared/app/nav.desktop.js

```diff
@@ -11,40 +11,47 @@ import {navigateTo, switchTo} from '../actions/route-tree'
 import {showUserProfile} from '../actions/profile'
 import {type Props, type StateProps, type DispatchProps, type OwnProps} from './nav'
 
-function Nav(props: Props) {
-  const visibleScreen = props.routeStack.findLast(r => !r.tags.layerOnTop)
-  if (!visibleScreen) {
-    throw new Error('no route component to render without layerOnTop tag')
+class Nav extends React.Component<Props> {
+  _switchTab = tab => {
```

Comment:
pass this._switchTab to children so they don't thrash as we remake this.props.switchTab